### ISSUE TITLE
LateInitializationError: Field 'annotationAst' has not been initialized

### ISSUE THREAD

**Issue by NaarGes** [dart-lang/sdk#47121]
I upgrade my packages using `dart pub outdated --mode=null-safety ` command. after updating my packages I got this error while running `dart migrate` to migrate my code to null-safety. also when I tried to run  `flutter pub run build_runner build --delete-conflicting-outputs` command for moor database.

I search for anything named `annotationAst` and there are only in the flutter itself path for example:
`/home/ns/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/analyzer-1.7.1/lib/src/summary2`

Why does this issue happen and How can I solve that?

> error
```
LateInitializationError: Field 'annotationAst' has not been initialized.
#0      ElementAnnotationImpl.annotationAst (package:analyzer/src/dart/element/element.dart)
#1      ConstantEvaluationEngine.computeDependencies (package:analyzer/src/dart/constant/evaluation.dart:336:39)
#2      _ConstantWalker._computeDependencies (package:analyzer/src/dart/constant/compute.dart:75:22)
#3      _ConstantNode.computeDependencies (package:analyzer/src/dart/constant/compute.dart:43:19)
#4      Node.getDependencies (package:analyzer/src/summary/link.dart:148:40)
#5      DependencyWalker.walk.strongConnect (package:analyzer/src/summary/link.dart:53:40)
#6      DependencyWalker.walk (package:analyzer/src/summary/link.dart:119:18)
#7      computeConstants (package:analyzer/src/dart/constant/compute.dart:26:14)
#8      LibraryAnalyzer._computeConstants (package:analyzer/src/dart/analysis/library_analyzer.dart:250:5)
#9      LibraryAnalyzer.analyzeSync (package:analyzer/src/dart/analysis/library_analyzer.dart:135:5)
#10     LibraryAnalyzer.analyze (package:analyzer/src/dart/analysis/library_analyzer.dart:103:12)
#11     AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1624:63)
#12     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:32:15)
#13     AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1600:20)
#14     AnalysisDriver._computeErrors (package:analyzer/src/dart/analysis/driver.dart:1678:26)
#15     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:1301:20)
#16     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2334:24)
<asynchronous suspension>
```

> dart version:
Dart SDK version: 2.13.4 (stable) (Wed Jun 23 13:08:41 2021 +0200) on "linux_x64"

> flutter version:
Flutter 2.2.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f4abaa0735 (10 weeks ago) • 2021-07-01 12:46:11 -0700
Engine • revision 241c87ad80
Tools • Dart 2.13.4

> flutter doctor -v
```[✓] Flutter (Channel stable, 2.2.3, on Linux, locale en_US.UTF-8)
    • Flutter version 2.2.3 at /home/ns/snap/flutter/common/flutter
    • Framework revision f4abaa0735 (10 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

⢿[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /home/ns/Android/Sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /snap/android-studio/114/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /snap/android-studio/114/android-studio
    • Flutter plugin version 60.1.2
    • Dart plugin version 203.8292
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 93.0.4577.63

• No issues found!
```

**Comment by NaarGes** [comment]
I have this issue in two different device. It's kind of unsolvable for me and I have time limits to migrate project
please help

**Comment by scheglov** [comment]
This looks like a bug in the analyzer. I would need an example of a project where it could be reproduced.

**Comment by NaarGes** [comment]
Sorry I couldn't send you the project because it wasn't for myself and I couldn't reproduce the error.
BTW the problem was solved after upgrading flutter to 2.5.1
Thank you
